packages/library-base/src/commonMain/kotlin/io/realm/kotlin/log/RealmLog.kt

```diff
@@ -0,0 +1,365 @@
+    }
+
+    /**
+     * Logs a [LogLevel.INFO] event.
+     *
+     * @param message optional message.
+     * @param args optional args used to format the message using a subset of `String.format`
+     * options. Only `%s`, `%d` and `%f` are supported.
+     */
+    public fun warn(message: String, vararg args: Any?) {
+        doLog(LogLevel.WARN, null, message, *args)
+    }
+
+    /**
+     * Logs a [LogLevel.ERROR] event.
+     *
+     * @param throwable optional exception to log.
+     */
+    public fun error(throwable: Throwable?) {
+        doLog(LogLevel.ERROR, throwable, null)
+    }
+
+    /**
+     * Logs a [LogLevel.ERROR] event.
+     *
+     * @param throwable optional exception to log.
+     * @param message optional message.
+     * @param args optional args used to format the message using a subset of `String.format`
+     * options. Only `%s`, `%d` and `%f` are supported.
+     */
+    public fun error(throwable: Throwable?, message: String, vararg args: Any?) {
+        doLog(LogLevel.ERROR, throwable, message, *args)
+    }
+
+    /**
+     * Logs a [LogLevel.ERROR] event.
+     *
+     * @param message optional message.
+     * @param args optional args used to format the message using a subset of `String.format`
+     * options. Only `%s`, `%d` and `%f` are supported.
+     */
+    public fun error(message: String, vararg args: Any?) {
+        doLog(LogLevel.ERROR, null, message, *args)
+    }
+
+    /**
+     * Logs a [LogLevel.WTF] event.
+     *
+     * @param throwable optional exception to log.
+     */
+    public fun wtf(throwable: Throwable?) {
+        doLog(LogLevel.WTF, throwable, null)
+    }
+
+    /**
+     * Logs a [LogLevel.WTF] event.
+     *
+     * @param throwable optional exception to log.
+     * @param message optional message.
+     * @param args optional args used to format the message using a subset of `String.format`
+     * options. Only `%s`, `%d` and `%f` are supported.
+     */
+    public fun wtf(throwable: Throwable?, message: String, vararg args: Any?) {
+        doLog(LogLevel.WTF, throwable, message, *args)
+    }
+
+    /**
+     * Logs a [LogLevel.WTF] event.
+     *
+     * @param message optional message.
+     * @param args optional args used to format the message using a subset of `String.format`
+     * options. Only `%s`, `%d` and `%f` are supported.
+     */
+    public fun wtf(message: String, vararg args: Any?) {
+        doLog(LogLevel.WTF, null, message, *args)
+    }
+
+    private fun doLog(level: LogLevel, throwable: Throwable?, message: String?, vararg args: Any?) {
+        if (level.priority >= this.level.priority) {
+            // Copy the reference to loggers so they are stable while iterating them.
+            val loggers = this.loggers
+            loggers.forEach {
+                it.log(level, throwable, message, *args)
+            }
+        }
+    }
+
+    private fun LogLevel.toCoreLogLevel(): CoreLogLevel {
```

Comment:
Seems very against our normal approach to have the implementation of this here. Isn't it an interop-thing?